### DISCUSSION TITLE
CORS issue

### DISCUSSION THREAD

**Discussion by emulienfou** [MrRefactoring/confluence.js#11] [Q&A]
Hi there!
Didn't want to open an issue, since I do not know if it's a bug or not.

I currently use Next.js, on server (NodeJS) side, everything works has expected.
However on client side, I received this error message:
`Access to XMLHttpRequest at 'https://example.atlassian.net/wiki/rest/api/content?spaceKey=AA&expand=ancestors,body.view,descendants.page' from origin 'http://localhost:3001' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

I also use your other library jira.js and it works properly on client side.
Maybe this is related to a configuration to Axios, or something else.

Do you have any idea what the problem could be?
Thanks

**Comment by MrRefactoring** [comment]
Hello David. I think you call some your endpoint (like localhost:7000/getContent) from client side and get this error?

Best wishes,
Vlad

**Reply by emulienfou** [reply]
Indeed, I get this error only in Browser side, but not in server side.

**Reply by MrRefactoring** [reply]
It's logically. You have setuped CORS policy in Next.js side?

**Comment by MrRefactoring** [comment]
And I think this article will be helpful for you: https://nextjs.org/docs/api-routes/api-middlewares

**Reply by emulienfou** [reply]
Yes I could create an API middleware that is used as a proxy to make the call to Concluence like as if i was in server side.
But I think this will be just a bandage and other people may have the same issues in the future!

**Reply by MrRefactoring** [reply]
Yeah. It's security behavior in browsers. Everyone should manage CORS policy by himself. And also I cannot impact on your server config.

Conclusion: you should manage your CORS policy in your server by yourself :)

**Reply by MrRefactoring** [reply]
Please let me know if you faced with any problems
